invenio_app_rdm/upgrade_scripts/migrate_11_0_to_12_0.py

```diff
@@ -94,12 +96,17 @@ def update_parent(record):
                     record.parent, pids
                 )
                 record.parent["pids"] = pids
-                current_rdm_records.records_service.pids.register_or_update(
-                    id_=record["id"],
-                    identity=system_identity,
-                    scheme="doi",
-                    parent=True,
-                )
+                # Have to commit here otherwise register_or_update won't get
+                # the above data
+                record.parent.commit()
```

Comment:
In particular without that, records with external DOI will cause register_or_update below to fail since RDMRecord is fetched independently in that method and therefore doesn' provide the right provider(!).